### ISSUE TITLE
[Windows/arm] crossgen System.Private.CoreLib fails with assert

### ISSUE THREAD

**Issue by jkotas** [dotnet/coreclr#17191]
From https://ci.dot.net/job/dotnet_coreclr/job/master/job/armlb_cross_checked_windows_nt_innerloop_prtest/513/consoleFull#-2059264786f66db340-b3cd-4e23-8190-9b52cb916333 :

```
Assert failure(PID 2548 [0x000009f4], Thread: 1308 [0x051c]): Assertion failed 'indirectCellAddress->IsCnsIntOrI() && indirectCellAddress->gtRegNum == REG_R2R_INDIRECT_PARAM' in '<>f__AnonymousType0`1[__Canon][System.__Canon]:ToString():ref:this' (IL size 77)
```

**Comment by jkotas** [comment]
cc @dotnet/jit-contrib  @dotnet/arm32-contrib

**Comment by BruceForstall** [comment]
All armlb jobs are failing. Proposed (possibly temporary) disabling of armlb: https://github.com/dotnet/coreclr/pull/17151

**Comment by jashook** [comment]
This is not fixed; however, the job is now disabled #17151.

**Comment by erozenfeld** [comment]
@BruceForstall @jashook Should this be closed since we are planning on doing #17490 soon?
